### PR TITLE
[front] - chore(tree): empty folders in trees 

### PR DESCRIPTION
**Pull request by JulesBelveze**
## Description

This PR addresses the handling of empty folders in the `DataSourceViewResourceSelectorTree` component. It introduces a new `Tree.Empty` component to display a "No documents" message for empty folders, improving the user experience when navigating through the data source tree.

**References:**
- https://github.com/dust-tt/tasks/issues/1359

## Risk

None

## Deploy Plan

Deploy `front`

### CODE REVIEW

File: front/components/DataSourceViewResourceSelectorTree.tsx [hunk 1]
```diff
@@ -120,6 +120,8 @@ function DataSourceViewResourceSelectorChildren({
   }
 
   const isTablesView = viewType === "tables";
+  const isEmptyManagedDataSource =
+    dataSourceView.category === "managed" && nodes.length === 0 && !parentId;
```
Comment (Fraggle) [inline front/components/DataSourceViewResourceSelectorTree.tsx:124]: the `!parentId` is to make sure we are at the root of the datasource ?

**Review by Fraggle — APPROVED** [review]
Lovely

File: front/components/DataSourceViewResourceSelectorTree.tsx (hunk 1 above)
Comment (JulesBelveze) [reply front/components/DataSourceViewResourceSelectorTree.tsx:124]: Yessir !

File: front/components/DataSourceViewResourceSelectorTree.tsx [hunk 2]
```diff
@@ -219,17 +221,16 @@ function DataSourceViewResourceSelectorChildren({
...
+        {isEmptyManagedDataSource && !readonly && (
+          <div className="flex w-full flex-col items-center gap-2 rounded-lg border bg-structure-50 py-2">
+            <span className="text-element-700">The Vault is empty!</span>
+            <RequestOrAddDataFromDataSourceModal
+              owner={owner}
+              dataSource={dataSourceView.dataSource}
+            />
+          </div>
+        )}
+        {parentId && nodes.length === 0 && <Tree.Empty label="No documents" />}
```
[the remark this answers is not in the document]
Comment (JulesBelveze) [reply front/components/DataSourceViewResourceSelectorTree.tsx:233]: We can't expand document but I'm wondering if a table with no sheet is possible? 🤔

**Pull request merged**